drivers/regulator/regulator_axp192.c

```diff
@@ -0,0 +1,438 @@
+};
+
+static const struct regulator_axp192_desc dcdc2_desc = {
+	.enable_reg = AXP192_REG_DCDC123_LDO23_CONTROL,
+	.enable_mask = AXP192_DCDC2_ENABLE_REG_MASK,
+	.enable_val = AXP192_DCDC2_ENABLE_REG_VAL,
+	.vsel_reg = AXP192_REG_DCDC2_VOLTAGE,
+	.vsel_mask = AXP192_DCDC2_VOLTAGE_REG_MASK,
+	.vsel_bitpos = AXP192_DCDC2_VOLTAGE_REG_BITPOS,
+	.max_ua = AXP192_DCDC2_CURRENT_MAX_UA,
+	.ranges = dcdc2_ranges,
+	.num_ranges = ARRAY_SIZE(dcdc2_ranges),
+};
+
+static struct linear_range dcdc3_ranges[] = {
+	LINEAR_RANGE_INIT(0u, 0u, 0u, 0u),
+	LINEAR_RANGE_INIT(AXP192_DCDC3_VOLTAGE_MIN_UV, AXP192_DCDC3_VOLTAGE_STEP_UV,
+			  AXP192_DCDC3_VOLTAGE_RANGE_MIN, AXP192_DCDC3_VOLTAGE_RANGE_MAX),
+};
+
+static const struct regulator_axp192_desc dcdc3_desc = {
+	.enable_reg = AXP192_REG_DCDC123_LDO23_CONTROL,
+	.enable_mask = AXP192_DCDC3_ENABLE_REG_MASK,
+	.enable_val = AXP192_DCDC3_ENABLE_REG_VAL,
+	.vsel_reg = AXP192_REG_DCDC3_VOLTAGE,
+	.vsel_mask = AXP192_DCDC3_VOLTAGE_REG_MASK,
+	.vsel_bitpos = AXP192_DCDC3_VOLTAGE_REG_BITPOS,
+	.max_ua = AXP192_DCDC3_CURRENT_MAX_UA,
+	.ranges = dcdc3_ranges,
+	.num_ranges = ARRAY_SIZE(dcdc3_ranges),
+};
+
+static struct linear_range ldo2_ranges[] = {
+	LINEAR_RANGE_INIT(0u, 0u, 0u, 0u),
+	LINEAR_RANGE_INIT(AXP192_LDO2_VOLTAGE_MIN_UV, AXP192_LDO2_VOLTAGE_STEP_UV,
+			  AXP192_LDO2_VOLTAGE_RANGE_MIN, AXP192_LDO2_VOLTAGE_RANGE_MAX),
+};
+
+static const struct regulator_axp192_desc ldo2_desc = {
+	.enable_reg = AXP192_REG_DCDC123_LDO23_CONTROL,
+	.enable_mask = AXP192_LDO2_ENABLE_REG_MASK,
+	.enable_val = AXP192_LDO2_ENABLE_REG_VAL,
+	.vsel_reg = AXP192_REG_LDO23_VOLTAGE,
+	.vsel_mask = AXP192_LDO2_VOLTAGE_REG_MASK,
+	.vsel_bitpos = AXP192_LDO2_VOLTAGE_REG_BITPOS,
+	.max_ua = AXP192_LDO2_CURRENT_MAX_UA,
+	.ranges = ldo2_ranges,
+	.num_ranges = ARRAY_SIZE(ldo2_ranges),
+};
+
+static struct linear_range ldo3_ranges[] = {
+	LINEAR_RANGE_INIT(0u, 0u, 0u, 0u),
+	LINEAR_RANGE_INIT(AXP192_LDO3_VOLTAGE_MIN_UV, AXP192_LDO3_VOLTAGE_STEP_UV,
+			  AXP192_LDO3_VOLTAGE_RANGE_MIN, AXP192_LDO3_VOLTAGE_RANGE_MAX),
+};
+
+static const struct regulator_axp192_desc ldo3_desc = {
+	.enable_reg = AXP192_REG_DCDC123_LDO23_CONTROL,
+	.enable_mask = AXP192_LDO3_ENABLE_REG_MASK,
+	.enable_val = AXP192_LDO3_ENABLE_REG_VAL,
+	.vsel_reg = AXP192_REG_LDO23_VOLTAGE,
+	.vsel_mask = AXP192_LDO3_VOLTAGE_REG_MASK,
+	.vsel_bitpos = AXP192_LDO3_VOLTAGE_REG_BITPOS,
+	.max_ua = AXP192_LDO3_CURRENT_MAX_UA,
+	.ranges = ldo3_ranges,
+	.num_ranges = ARRAY_SIZE(ldo3_ranges),
+};
+
+int axp192_enable(const struct device *dev)
+{
+	const struct regulator_axp192_config *config = dev->config;
+	const struct regulator_axp192_common_config *cconfig = config->mfd->config;
+	int ret = 0;
+
+	LOG_INST_DBG(config->log, "Enabling regulator");
+	LOG_INST_DBG(config->log, "[0x%02x]=0x%02x mask=0x%02x", (uint32_t)config->desc->enable_reg,
+		     (uint32_t)config->desc->enable_val, (uint32_t)config->desc->enable_mask);
```

Comment:
is instance debug really needed? This is rarely used in devices (also, there's tipically 1 instance of regulators, even though we support N)

Reply:
There is an instance for every Buck. That was the easiest solution I found to directly identify in the debug log, to which DCDC the debug message relates to.
Is there something bad with instance logging?